src/db/sysdb_subdomains.c

```diff
@@ -218,6 +225,13 @@ check_subdom_config_file(struct confdb_ctx *confdb,
           sd_conf_path, CONFDB_DOMAIN_FQ,
           subdomain->fqnames ? "TRUE" : "FALSE");
 
+    /* use fully qualified names as output in order to avoid causing
+     * conflicts with users who have the same name and either the
+     * shortname user resolution is enabled or the trusted domain has
+     * been explicitly set to use non-fully qualified names as input.
+     */
+    subdomain->output_fqnames = true;
+
```

Comment:
Hmm. You're right.

When I touched this part of the code I was with `use_fully_qualified_name = False` in my mind. But that's actually not related.

I'll remove this piece when updating the patch.